backend/src/main/resources/schema.graphqls

```diff
@@ -357,8 +360,8 @@ type Mutation {
     street: String!
     streetTwo: String
     city: String
-    state: String!
-    zipCode: String!
+    state: String! @length(min: 2, max: 2)
+    zipCode: String! @length(min: 5, max: 10)
```

Comment:
@benwarfield-usds @jeskew-gov I'm taking over the PR while @brmaden is OOO.

Any concerns with using directives here before I continue down this path? (E.g. jeskew _was_ going to use this approach for permissions but decided against it.)

Reply:
Directives are fine here. This PR is actually using the approach I adopted for permissions; I had previously been using the same instrumentation-based approach as https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/analysis/MaxQueryDepthInstrumentation.java, but that turned out not to play well with SR audit logging.

Reply:
I have not been able to figure out a way to modify the input data in an argument directive before the data fetcher is called.  This would be necessary to support things like trimming strings, converting empty strings to null, or forcing case so we wouldn't have to do that in the resolvers.

If we are okay with only being able to do validation, we should consider just using `graphql-java-extended-validation` as originally suggested.  The library has other directives, like `@Pattern` that supports validating against a regex so would also address https://github.com/CDCgov/prime-simplereport/issues/997

Reply:
@TimBest @benwarfield-usds 

Thoughts? 
Would it be ok with this approach: FE limits field lengths and the backend enforces (throws error) if length constraints are violated.

Reply:
^ thats the idea

Reply:
Well.. it could clip too-long strings, but yeah... I see your point about too-short strings.

FYI I think we should probably have an absolute max for strings that don't have some specification... like 256? That can then be overridden by @lenth(min: 0, max: 512)

Seems like part of this changelist should probably include figuring out min/max for all required strings?